models/stocks.js

```diff
@@ -38,7 +40,84 @@ const fetchStocks = async () => {
   }
 }
 
+/**
+ * Fetches the user stocks
+ * @return {Promise<userStocks|object>}
+ */
+const fetchUserStocks = async (userId, stockId = null) => {
+  try {
+    let userStocksRef = ''
+    if (stockId) {
+      userStocksRef = await userStocksModel.where('userId', '==', userId).where('stockId', '==', stockId).get()
+      const [userStocks] = userStocksRef.docs
+      if (userStocks) {
+        return { id: userStocks.id, ...userStocks.data() }
+      }
+      return {}
+    }
+
+    userStocksRef = await userStocksModel.where('userId', '==', userId).get()
+    const userStocks = []
+    userStocksRef.forEach((stock) => {
+      userStocks.push({
+        id: stock.id,
+        ...stock.data()
+      })
```

Comment:
Lot of repetition, can this be simplified?

Reply:
Yes, I am finding a way to making it simple.

Reply:
https://github.com/Real-Dev-Squad/website-backend/pull/265/commits/b2cd138a77efde53e934a063fd6f021608731fcf